### PR TITLE
Add fetching content from content_url

### PR DESCRIPTION
**Pull request by ErebusZ**

### CODE REVIEW

File: agent/utils.py [hunk 1]
```diff
@@ -138,3 +148,49 @@ def get_file_type(content: bytes, path: str | None) -> str:
...
+    """Download a file.
+
+    Args:
+        file_url : The URL of the file to download.
+
+    Returns:
+        bytes: The content of the file.
+    """
+    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT)
+    if response.status_code == 200 and isinstance(response.content, bytes):
```
Comment (PiranhaSa) [inline agent/utils.py:170]:
```suggestion
    if response.status_code == 200 and isinstance(response.content, bytes) is True:
```

File: agent/utils.py [hunk 2]
```diff
@@ -138,3 +148,49 @@ def get_file_type(content: bytes, path: str | None) -> str:
...
+    """Get the file content from a message.
+
+    Args:
+        message : The message containing the file data.
+
+    Returns:
+        bytes: The content of the file.
+    """
+    content = message.data.get("content")
+    if content is not None and isinstance(content, bytes):
```
Comment (PiranhaSa) [inline agent/utils.py:186]:
```suggestion
    if content is not None and isinstance(content, bytes) is True:
```

File: agent/utils.py [hunk 3]
```diff
@@ -138,3 +148,49 @@ def get_file_type(content: bytes, path: str | None) -> str:
...
+def _download_file(file_url: str) -> bytes | None:
+    """Download a file.
+
+    Args:
+        file_url : The URL of the file to download.
+
+    Returns:
+        bytes: The content of the file.
+    """
+    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT)
```
Comment (BlueSquare1) [inline agent/utils.py:169]:
```suggestion
    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT, verify=False)
```

**Review by BlueSquare1 — CHANGES_REQUESTED** [review]

File: agent/utils.py [hunk 4]
```diff
@@ -138,3 +148,49 @@ def get_file_type(content: bytes, path: str | None) -> str:
...
+        file_url : The URL of the file to download.
+
+    Returns:
+        bytes: The content of the file.
+    """
+    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT, verify=False)
+    if response.status_code == 200 and isinstance(response.content, bytes) is True:
+        return response.content
+
+    return None
```
Comment (3asm) [inline agent/utils.py:173]:
```suggestion
    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT)
    if response.status_code == 200 and isinstance(response.content, bytes) is True:
        return response.content

    return None
```

Not here.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**